TESTS/mbed_drivers/lp_timeout/main.cpp

```diff
@@ -59,7 +59,10 @@ void lp_timeout_1s_deepsleep(void)
      */
     timestamp_t start = lp_ticker_read();
     lpt.attach(&cb_done, 1);
-    deepsleep();
+    /* Make sure deepsleep is allowed, to go to deepsleep */
+    bool deep_sleep_allowed = sleep_manager_can_deep_sleep();
+    TEST_ASSERT_TRUE_MESSAGE(deep_sleep_allowed, "Deep sleep should be allowed");
+    sleep();
```

Comment:
this is assuming that using LowPower object does allow deepsleep, and using sleep() should go into deepsleep, but would it? :-) As this test is for drivers, this looks fine, what about the below test for hal? Same? no direct deepsleep hal call ?

Reply:
I was speaking about lp_timeout_1s_sleep function.

Reply:
> I was speaking about lp_timeout_1s_sleep function.

Should have lock there.